### PR TITLE
[RM Ch04] Underlay and Overlay Networking concepts

### PR DESCRIPTION
**Pull request by TFredberg**
Added an updated Underlay and Overlay Networking concepts from the Baldy release approved Tech/approach.md

### CODE REVIEW

File: doc/ref_model/chapters/chapter04.md [hunk 1]
```diff
@@ -623,6 +623,20 @@ Before we start developing a deep model we need to agree on some foundational co
  - Software Defined Underlay and Overlay Networking concepts 
  - Programmable Networking Fabric concept
 
+<a name="4.3.1.1"></a>
+#### 4.3.1.1 Underlay and Overlay Networking concepts
+
+The ETSI NFV model divide networking in an Underlay and an Overlay Network layer. The purpose with this layering is to ensure separation of the Virtualisation Tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network in between all Ethernet connected hardware (HW) devices.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter04.md:629]:
```suggestion
The ETSI NFV model divides networking into an Underlay and an Overlay Network layer. The purpose of this layering is to ensure separation of the Virtualisation Tenants (Workload) Overlay Networks from each other, whilst allowing the traffic to flow on the shared Underlay Network (connected hardware (HW) devices).
```

File: doc/ref_model/chapters/chapter04.md [hunk 2]
```diff
@@ -623,6 +623,20 @@ Before we start developing a deep model we need to agree on some foundational co
...
+
+The Overlay Networking separation is often done through encapsulation of Tenants traffic using overlay protocols e.g. through VxLAN or EVPN on the Underlay Networks e.g. based on L2 (VLAN) or L3 (IP) networks.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter04.md:631]: two "e.g." makes it confusing
```suggestion
The Overlay Networking separation is often done through encapsulation of Tenant traffic using overlay protocols such as VxLAN or EVPN over the Underlay Networks (e.g. L2 (VLAN) or L3 (IP) networks).
```

File: doc/ref_model/chapters/chapter04.md [hunk 3]
```diff
@@ -623,6 +623,20 @@ Before we start developing a deep model we need to agree on some foundational co
...
+
+In some instances, the Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively take over the Physical and Virtual Functions of the NIC directly into the VNF/CNF Tenant. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter04.md:633]:
```suggestion
In some instances, the Virtualisation Tenants can bypass the Overlay Networking encapsulation to achieve better performance or network visibility/control. A common method to bypass the Overlay Networking encapsulation normally done by the Virtualisation Layer, is the VNF/CNF usage of SR-IOV that effectively directly connects the Physical and Virtual Functions of the NIC to the Tenant VNF/CNF. In these cases, the Underlay Networking must handle the separation e.g. through a Virtual Termination End Point (VTEP) that encapsulate the Overlay Network traffic.
```

File: doc/ref_model/chapters/chapter04.md [hunk 4]
```diff
@@ -623,6 +623,20 @@ Before we start developing a deep model we need to agree on some foundational co
...
+
+> **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and available standards. Until suitable technologies and standards are developed, CNTT have a set of agreed exemptions that forces the Underlay Networking to handle the bypassed Overlay Networking separation.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter04.md:635]:
```suggestion
> **Note:** Bypassing the Overlay Networking layer is a violation of the basic CNTT decoupling principles, but in some cases unavoidable with existing technologies and available standards. Until suitable technologies and standards are developed, CNTT has specified a set of exemptions that allows the Underlay Networking to handle the bypassed Overlay Networking separation.
```

File: doc/ref_model/chapters/chapter04.md [hunk 5]
```diff
@@ -623,6 +623,20 @@ Before we start developing a deep model we need to agree on some foundational co
...
+
+VTEP could be manually provisioned in the Underlay Networking or be automated and controlled through a Software Defined Networking controller interfaces into the underlying networking in the HW Infrastructure Layer.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter04.md:637]:
```suggestion
VTEP could be manually provisioned in the Underlay Networking or be automatically provisioned and managed through Software Defined Networking (SDN) controller interfaces into the underlying networking in the HW Infrastructure Layer. 
```

**Comment by rabi-abdel** [comment]
will merge this as this will go to #1771

**Pull request merged**